app/helpers/application_helper.rb

```diff
@@ -225,18 +225,27 @@ def edit_select(object, property, options = {})
 
   def flot_pie_chart(name, title, data, options = {})
     data = data.map { |k, v| {:label => k.to_s.humanize, :data => v} } if data.is_a?(Hash)
-    data.map { |element| element[:label] = truncate(element[:label], :length => 16) }
-    header = content_tag(:h4, options[:show_title] ? title : '', :class => 'ca pie-title', :'data-original-title' => _("Expand the chart"), :rel => 'twipsy')
-    link_to_function(header, "expand_chart(this)") +
-        content_tag(:div, nil,
-          { :id    => name,
-            :class => 'statistics-pie',
-            :data  => {
-              :title  => title,
-              :series => data,
-              :url    => options[:search] ? "#{request.script_name}/hosts?search=#{URI.encode(options.delete(:search))}" : "#",
-            },
-          }.merge(options))
+    Foreman::Deprecation.deprecation_warning('3.0', '#flot_pie_chart is by default rendered by react Donut chart with default configuration. You can still opt-in the old behaviour by option "use_legacy_flot", but please prefer to use the React component directly.')
+    if options[:use_legacy_flot]
+      data.map { |element| element[:label] = truncate(element[:label], :length => 16) }
+      header = content_tag(:h4, options[:show_title] ? title : '', :class => 'ca pie-title', :'data-original-title' => _("Expand the chart"), :rel => 'twipsy')
+      link_to_function(header, "expand_chart(this)") +
+          content_tag(:div, nil,
+            { :id    => name,
+              :class => 'statistics-pie',
+              :data  => {
+                :title  => title,
+                :series => data,
+                :url    => options[:search] ? "#{request.script_name}/hosts?search=#{URI.encode(options.delete(:search))}" : "#",
+              },
+            }.merge(options))
+    else
+      react_component('ChartBox', type: 'donut',
+                                  status: 'RESOLVED',
+                                  title: title,
+                                  data: data,
```

Comment:
tested with https://github.com/theforeman/foreman_omaha/pull/55 and I think it should be
```suggestion
                                  chart: {
                                    data: data
                                  },
```